module/zfs/vdev.c

```diff
@@ -1642,6 +1642,17 @@ vdev_open(vdev_t *vd)
 
 	error = vd->vdev_ops->vdev_op_open(vd, &osize, &max_osize, &ashift);
 
+	/*
+	 * Physical volume size should never be larger than its max size, unless
+	 * the disk has shrunk while we were reading it or the device is buggy
+	 * or damaged: either way it's not safe for use, bail out of the open.
+	 */
+	if (osize > max_osize) {
+		vdev_set_state(vd, B_TRUE, VDEV_STATE_CANT_OPEN,
+		    VDEV_AUX_OPEN_FAILED);
+		return (SET_ERROR(EOVERFLOW));
```

Comment:
Rather than add `EOVERFLOW` to `zpool_standard_error_fmt()` why don't we return `ENXIO` here instead to maintain compatibility.  `EOVERFLOW` is more descriptive, but for this very unlikely case I think the generic error, "one or more devices is currently unavailable", would be acceptable.